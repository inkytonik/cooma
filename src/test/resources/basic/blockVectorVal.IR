letc k1 x = letc k2 y = letc k3 ab = letc k4 z = $halt z
                                     letv r6 = concat x y
                                     letv f9 = select x get
                                     letv i10 = 0
                                     letc k7 r8 = letv f13 = select x append
                                                  letv i14 = 989
                                                  letc k11 r12 = letv f19 = select x put
                                                                 letv i20 = 0
                                                                 letc k17 r18 = letv i21 = 151
                                                                                letc k15 r16 = letv f26 = select ab slice
                                                                                               letv i27 = 0
                                                                                               letc k24 r25 = letv i28 = 4
                                                                                                              letc k22 r23 = letv r5 = { a = r6, b = r8, c = r12, d = r16, e = r23 }
                                                                                                                             k4 r5
                                                                                                              r25 k22 i28
                                                                                               f26 k24 i27
                                                                                r18 k15 i21
                                                                 f19 k17 i20
                                                  f13 k11 i14
                                     f9 k7 i10
                        letv i30 = 1
                        letv i31 = 2
                        letv i32 = 3
                        letv i33 = 4
                        letv i34 = 5
                        letv v29 = [i30,i31,i32,i33,i34]
                        k3 v29
            letv i36 = 555
            letv v35 = [i36]
            k2 v35
letv i38 = 999
letv v37 = [i38]
k1 v37