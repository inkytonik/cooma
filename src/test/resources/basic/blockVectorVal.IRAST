LetC(
    "k1",
    "x",
    LetC(
        "k2",
        "y",
        LetC(
            "k3",
            "ab",
            LetC(
                "k4",
                "z",
                AppC(
                    "$halt",
                    "z"),
                LetV(
                    "r6",
                    PrmV(
                        ConcatP(),
                        Vector(
                            "x",
                            "y")),
                    LetV(
                        "f9",
                        PrmV(
                            SelectP(),
                            Vector(
                                "x",
                                "get")),
                        LetV(
                            "i10",
                            IntV(
                                0),
                            LetC(
                                "k7",
                                "r8",
                                LetV(
                                    "f13",
                                    PrmV(
                                        SelectP(),
                                        Vector(
                                            "x",
                                            "append")),
                                    LetV(
                                        "i14",
                                        IntV(
                                            989),
                                        LetC(
                                            "k11",
                                            "r12",
                                            LetV(
                                                "f19",
                                                PrmV(
                                                    SelectP(),
                                                    Vector(
                                                        "x",
                                                        "put")),
                                                LetV(
                                                    "i20",
                                                    IntV(
                                                        0),
                                                    LetC(
                                                        "k17",
                                                        "r18",
                                                        LetV(
                                                            "i21",
                                                            IntV(
                                                                151),
                                                            LetC(
                                                                "k15",
                                                                "r16",
                                                                LetV(
                                                                    "f26",
                                                                    PrmV(
                                                                        SelectP(),
                                                                        Vector(
                                                                            "ab",
                                                                            "slice")),
                                                                    LetV(
                                                                        "i27",
                                                                        IntV(
                                                                            0),
                                                                        LetC(
                                                                            "k24",
                                                                            "r25",
                                                                            LetV(
                                                                                "i28",
                                                                                IntV(
                                                                                    4),
                                                                                LetC(
                                                                                    "k22",
                                                                                    "r23",
                                                                                    LetV(
                                                                                        "r5",
                                                                                        RecV(
                                                                                            Vector(
                                                                                                FieldValue(
                                                                                                    "a",
                                                                                                    "r6"),
                                                                                                FieldValue(
                                                                                                    "b",
                                                                                                    "r8"),
                                                                                                FieldValue(
                                                                                                    "c",
                                                                                                    "r12"),
                                                                                                FieldValue(
                                                                                                    "d",
                                                                                                    "r16"),
                                                                                                FieldValue(
                                                                                                    "e",
                                                                                                    "r23"))),
                                                                                        AppC(
                                                                                            "k4",
                                                                                            "r5")),
                                                                                    AppF(
                                                                                        "r25",
                                                                                        "k22",
                                                                                        "i28"))),
                                                                            AppF(
                                                                                "f26",
                                                                                "k24",
                                                                                "i27")))),
                                                                AppF(
                                                                    "r18",
                                                                    "k15",
                                                                    "i21"))),
                                                        AppF(
                                                            "f19",
                                                            "k17",
                                                            "i20")))),
                                            AppF(
                                                "f13",
                                                "k11",
                                                "i14")))),
                                AppF(
                                    "f9",
                                    "k7",
                                    "i10")))))),
            LetV(
                "i30",
                IntV(
                    1),
                LetV(
                    "i31",
                    IntV(
                        2),
                    LetV(
                        "i32",
                        IntV(
                            3),
                        LetV(
                            "i33",
                            IntV(
                                4),
                            LetV(
                                "i34",
                                IntV(
                                    5),
                                LetV(
                                    "v29",
                                    VecV(
                                        Vector(
                                            "i30",
                                            "i31",
                                            "i32",
                                            "i33",
                                            "i34")),
                                    AppC(
                                        "k3",
                                        "v29")))))))),
        LetV(
            "i36",
            IntV(
                555),
            LetV(
                "v35",
                VecV(
                    Vector(
                        "i36")),
                AppC(
                    "k2",
                    "v35")))),
    LetV(
        "i38",
        IntV(
            999),
        LetV(
            "v37",
            VecV(
                Vector(
                    "i38")),
            AppC(
                "k1",
                "v37"))))