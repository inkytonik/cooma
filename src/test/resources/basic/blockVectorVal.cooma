{
    val x = [999]
    val y = [555]
    val ab = [1,2,3,4,5]
    val z = {
              a = x & y,
              b = x.get(0),
              c = x.append(989),
              d = x.put(0,151),
              e = ab.slice(0,4)
             }
    z
}
