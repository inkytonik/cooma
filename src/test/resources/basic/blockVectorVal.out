{ a = [999,555], b = 999, c = [999,989], d = [151], e = [1,2,3,4] }
